enums.rb

```diff
@@ -0,0 +1,128 @@
+# rubocop:disable Metrics/ModuleLength
+module Enumerable
+  # rubocop:enable Metrics/ModuleLength
+  # rubocop:disable Metrics/CyclomaticComplexity, Metrics/PerceivedComplexity
+
+  def my_each
+    return to_enum(:my_each) unless block_given?
+
+    size.times do |i|
+      yield to_a[i]
+    end
+    self
+  end
+
+  def my_each_with_index
+    return to_enum(:my_each) unless block_given?
+
+    size.times do |i|
+      yield to_a[i], i
+    end
+    self
+  end
+
+  def my_all?(arg = nil)
+    if block_given? && !arg.nil?
+      return false unless my_all?(arg) && my_all?
+    end
+    if block_given?
+      my_each { |i| return false unless yield(i) }
+    elsif arg.nil?
+      my_each { |i| return false unless i }
+    elsif arg.class == Class
+      my_each { |i| return false unless i.class <= arg }
+    elsif arg.class == Regexp
+      my_each { |i| return false unless i =~ arg }
+    else
+      my_each { |i| return false unless i == arg && i.class <= arg.class }
+    end
+
+    true
+  end
+
+  def my_select
+    return to_enum(:my_select) unless block_given?
+
+    new_array = []
+    my_each do |i|
+      new_array.push(i) if yield(i)
+    end
+    new_array
+  end
+
+  def my_any?(arg = nil)
+    if block_given? && !arg.nil?
+      return true unless my_any?(arg) && my_any?
+    end
+
+    if block_given?
+      my_each { |i| return true unless yield(i) }
+    elsif arg.nil?
+      my_each { |i| return true unless i }
+    elsif arg.class == Class
+      my_each { |i| return true unless i.class <= arg }
+    elsif arg.class == Regexp
+      my_each { |i| return true unless i =~ arg }
+    else
+      my_each { |i| return true unless i == arg && i.class <= arg.class }
+    end
+
+    false
+  end
+
+  def my_none?(arg = nil)
+    my_each do |i|
+      if block_given?
+        return false if yield(i)
+      elsif arg.class == Regexp
+        my_each { return true unless i =~ arg }
+      elsif !i.nil? && i != false
+      else
+        return false
+      end
+    end
+    true
+  end
+
+  def my_count(arg = nil)
+    count = 0
+    each do |i|
+      if block_given?
+        count += 1 if yield(i) == true
+      elsif !arg.nil?
+        count += 1 if i == arg
+      else
+        count += 1
+      end
+    end
+    count
+  end
+
+  def my_map(&arg)
+    return to_enum(:my_map) unless block_given?
+
+    modified_array = []
+    my_each { |i| modified_array << arg.call(i) }
+
+    modified_array
+  end
```

Comment:
* [ ] should have working `my_map` identical and returns the same thing as ruby's [`map`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-map). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_map.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))

#### my_map 

- [ ] in the specifications of the project in the [Odin project](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks) # 12, you were asked to modify your my_map method to take either a proc or a block. And if both a proc and a block are given, only execute the proc. The real map doesn't have this feature, but you were asked to have it so let's work on it

Given 
```ruby
my_proc = Proc.new {|num| num > 10 }
```
Result
```ruby
puts [18, 22, 5, 6] .my_map(my_proc) {|num| num < 10 }  => true true false false
```